pkg/kubectl/cmd/set/set_limit.go

```diff
@@ -0,0 +1,222 @@
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "Filename, directory, or URL to a file identifying the resource to get from the server"
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
+	cmd.Flags().BoolVar(&options.All, "all", false, "select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on")
+	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", "*", "The names of containers in the selected pod templates to change - may use wildcards")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set limit will NOT contact api-server but run locally")
+	cmdutil.AddRecordFlag(cmd)
+	cmd.Flags().BoolVar(&options.Remove, "remove", false, "If true, set limit will remove resource limit/requests from pod template")
+	cmd.Flags().StringVar(&options.Limits, "limits", options.Limits, "Here is the limit usage stuff")
+	cmd.Flags().StringVar(&options.Requests, "requests", options.Requests, "Here is the limit usage stuff")
+	cmdutil.AddRecursiveFlag(cmd, &options.Recursive)
+	return cmd
+}
+
+func (o *LimitOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.Mapper, o.Typer = f.Object(false)
+	o.UpdatePodSpecForObject = f.UpdatePodSpecForObject
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = cmdutil.GetFlagString(cmd, "output") == "name"
+	o.Record = cmdutil.GetRecordFlag(cmd)
+	o.ChangeCause = f.Command()
+	o.PrintObject = f.PrintObject
+	o.Cmd = cmd
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	builder := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
+		Flatten()
+	if !o.Local {
+		builder = builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, args...).
+			Latest()
+	}
+
+	o.Infos, err = builder.Do().Infos()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *LimitOptions) Validate() error {
+	var err error
+	//if len(o.Limits) == 0 && len(o.Requests) == 0 && o.Remove == false {
+	if (len(o.Limits) == 0 && len(o.Requests) == 0) != o.Remove {
+		return fmt.Errorf("you must specify an update to limits or requests (in the form of --limits/--requests or --remove but not both)")
+	}
+
+	o.ResourceRequirements, err = kubectl.HandleResourceRequirements(map[string]string{"limits": o.Limits, "requests": o.Requests})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *LimitOptions) Run() error {
+	allErrs := []error{}
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		transformed := false
+		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *api.PodSpec) error {
```

Comment:
Other things may have resources besides pods (quota and limits).  We should make an argument for or against whether this command should be polymorphic to type.  The container selector is probably the main issue - we'd need a different selector for sub quota info.